pkg/metrics/aggregator.go

```diff
@@ -0,0 +1,114 @@
+package metrics
+
+import (
+	"sync"
+	"time"
+
+	v1 "github.com/openshift/api/config/v1"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	providerLabel = "provider"
+)
+
+type providerKey struct {
+	name      string
+	namespace string
+}
+
+type AdoptionMetricsAggregator struct {
+	identityProviders   *prometheus.GaugeVec
+	clusterAdmin        prometheus.Gauge
+	providerMap         map[providerKey][]v1.IdentityProviderType
+	mutex               sync.Mutex
+	aggregationInterval time.Duration
+}
+
+func NewMetricsAggregator(aggregationInterval time.Duration) *AdoptionMetricsAggregator {
+	collector := &AdoptionMetricsAggregator{
+		identityProviders: prometheus.NewGaugeVec(prometheus.GaugeOpts{
+			Name:        "identity_provider",
+			Help:        "Indicates if a identity provider is enabled",
+			ConstLabels: map[string]string{"name": "osd_exporter"},
+		}, []string{providerLabel}),
+		clusterAdmin: prometheus.NewGauge(prometheus.GaugeOpts{
+			Name: "cluster_admin_enabled",
+			Help: "Indicates if the cluster-admin role is enabled",
+		}),
+		providerMap:         make(map[providerKey][]v1.IdentityProviderType),
+		aggregationInterval: aggregationInterval,
+	}
+	collector.clusterAdmin.Set(0)
```

Comment:
Nit: Use `SetClusterAdmin` here?

Similar throughout: Do you want to use the class getters/setters instead of directly accessing the fields?